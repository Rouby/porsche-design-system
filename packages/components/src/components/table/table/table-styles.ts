import {
  getMediaQueryMin,
  spacingStaticLarge,
  spacingStaticMedium,
  spacingStaticSmall,
  textSmallStyle,
} from '@porsche-design-system/utilities-v2';
import { getCss } from '../../../utils';
import {
  addImportantToEachRule,
  getFocusJssStyle,
  getThemedColors,
  hostHiddenStyles,
  pxToRemWithUnit,
} from '../../../styles';

const { primaryColor } = getThemedColors('light');

export const getComponentCss = (): string => {
  return getCss({
    '@global': {
      ':host': addImportantToEachRule({
        display: 'block',
        ...hostHiddenStyles,
      }),
    },
    caption: {
      marginBottom: spacingStaticSmall,
      [getMediaQueryMin('m')]: {
        marginBottom: spacingStaticMedium,
      },
    },
    root: {
      position: 'relative',
    },
    'scroll-area': {
      overflow: 'auto visible',
      ...getFocusJssStyle({ offset: -1 }),
    },
    table: {
      position: 'relative',
      width: '100%',
      display: 'table',
      ...textSmallStyle,
      textAlign: 'left',
      color: primaryColor,
      whiteSpace: 'nowrap',
    },
    'scroll-trigger': {
      position: 'absolute',
      top: 0,
      right: '1px',
      width: '1px',
      height: '1px',
      visibility: 'hidden',
    },
    'scroll-indicator': {
      position: 'absolute',
      top: 0,
      right: 0,
      bottom: 0,
      paddingLeft: spacingStaticLarge,
      pointerEvents: 'none',
      display: 'flex',
      alignItems: 'center',
      background: 'linear-gradient(90deg, rgba(255, 255, 255, 0) 0%, rgba(255, 255, 255, 1) 50%)',
      '&::before': {
        content: '""',
        position: 'absolute',
        top: 0,
        right: 0,
        bottom: 0,
        width: pxToRemWithUnit(48),
        pointerEvents: 'auto',
      },
    },
    'scroll-button': {
      padding: pxToRemWithUnit(12),
      pointerEvents: 'auto',
    },
  });
};
