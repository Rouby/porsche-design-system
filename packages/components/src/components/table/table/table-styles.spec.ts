import { getComponentCss } from './table-styles';

describe('getComponentCss()', () => {
  it('should return correct css', () => {
    expect(getComponentCss()).toMatchSnapshot();
  });
});
