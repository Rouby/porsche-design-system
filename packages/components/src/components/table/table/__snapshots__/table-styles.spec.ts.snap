// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentCss() should return correct css 1`] = `
":host {
  display: block !important;
}
:host([hidden]) {
  display: none !important;
}
.caption {
  margin-bottom: 8px;
}
.root {
  position: relative;
}
.scroll-area {
  overflow: auto visible;
  outline: 1px solid transparent;
  outline-offset: -1px;
}
.scroll-area::-moz-focus-inner {
  border: 0;
}
.scroll-area:focus {
  outline-color: currentColor;
}
.scroll-area:focus:not(:focus-visible) {
  outline-color: transparent;
}
.table {
  position: relative;
  width: 100%;
  display: table;
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  text-align: left;
  color: #010205;
  white-space: nowrap;
}
.scroll-trigger {
  position: absolute;
  top: 0;
  right: 1px;
  width: 1px;
  height: 1px;
  visibility: hidden;
}
.scroll-indicator {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  padding-left: 32px;
  pointer-events: none;
  display: flex;
  align-items: center;
  background: linear-gradient(90deg, rgba(255, 255, 255, 0) 0%, rgba(255, 255, 255, 1) 50%);
}
.scroll-indicator::before {
  content: "";
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  width: 3rem;
  pointer-events: auto;
}
.scroll-button {
  padding: 0.75rem;
  pointer-events: auto;
}
@media(min-width:1000px) {
  .caption {
    margin-bottom: 16px;
  }
}
"
`;
